### ISSUE TITLE
Issue with swipe loop

### ISSUE THREAD

**Issue by jojo91** [cerberustesting/cerberus-source#1644]
Hi,

I'm currently trying to create a test that will swipe in a list and stop swipping on a specific text. Then click to the text to redirect to a detail view. It works but not in an efficient way.
Here is my test : 
<img width="845" alt="screen shot 2018-03-09 at 16 33 14" src="https://user-images.githubusercontent.com/3472541/37215250-b5e7377a-23b7-11e8-9b3f-b9e1ac59b0e1.png">

I don't understand why but only two swipe are needed to go on my element and the test does 20 swipe.

Anybody have an idea ?

**Comment by j0hnsa** [comment]
Hello,

on the conditions to execute the step wouldn't it be more logical to use something like this:

![imagem](https://user-images.githubusercontent.com/16339891/37218222-b67123e0-23b7-11e8-9c48-c47deb97378e.png)

I dont know how would a property would start to exist in a middle of the test. I believe the step is executing the maximum of the step loop.

Can you give it a try?

Best Regards

**Comment by j0hnsa** [comment]
the parameter cerberus_loopstep_max defaults to 20:

![imagem](https://user-images.githubusercontent.com/16339891/37218488-7e3641bc-23b8-11e8-939a-8d8c2d457207.png)

Regards

**Comment by jojo91** [comment]
Thank you for your answer but I had already tested with the condition "ifElementPresent" and it doesn't work. The swipe is not executed with the condition "ifElementPresent" (don't know why) but with "ifPropertyExist" it works but it swipe 20 times.

I'm not gonna change the cerberus_loopstep_max parameter because if I have an element very far in a list I maybe need to execute 20 swipe.

**Comment by j0hnsa** [comment]
Yes I dont think changing the parameter work's either. But with whileconditionFalse and ifElementPresent should work. I'll try it myself and share results.

Regards

**Comment by j0hnsa** [comment]
I've tried like suggested and it worked:

![imagem](https://user-images.githubusercontent.com/16339891/37220596-061cf1c4-23bf-11e8-842e-d91d36275a24.png)

![imagem](https://user-images.githubusercontent.com/16339891/37220621-15593ad0-23bf-11e8-8f5b-7842ff056e0b.png)

Hope it helps,

Regards

**Comment by jojo91** [comment]
Does it works if you don't specify the direction ?

**Comment by j0hnsa** [comment]
No, for android the swipe only works with custom (absolute) coordinates. For ios it works both with directions and (relative) coordinates.

**Comment by jojo91** [comment]
After a lot of testing I've made some discoveries.
First I thought that when I used ElmentIsPresent, nothing happen and I didn't knew why. In fact it works but it's very very slow. 90 seconds to check if the element was there. After changing "cerberus_appium_wait_element" and "cerberus_selenium_wait_element" value from 90 to 5 seconds, it is way faster now.

But I still have one issue. The loop is repeating until the element is present but sometimes you can have an element that is present but not visible and when you try to click on this element the test fail. How can I prevent this to happen ?

**Comment by j0hnsa** [comment]
Yes if you set the timeout on the execution page it will also work, if no timeout is set it uses the default of that parameter.

![imagem](https://user-images.githubusercontent.com/16339891/37462217-10fca16e-2849-11e8-82bb-acbc72e0eca5.png)

about the present/visible from my experience appium will consider present when the element is visible on the screen. Do you have any example where this is happening?

Regards

**Comment by jojo91** [comment]
Here is my example:

My swipping  test :

<img width="841" alt="screen shot 2018-03-15 at 14 14 42" src="https://user-images.githubusercontent.com/3472541/37465424-5b1232fc-285b-11e8-9b5c-7d7a4606b084.png">



The swipping execution : 

<img width="1085" alt="screen shot 2018-03-15 at 14 17 51" src="https://user-images.githubusercontent.com/3472541/37465567-c4c2d10c-285b-11e8-94a3-a602f75e2895.png">

<img width="1083" alt="screen shot 2018-03-15 at 14 18 00" src="https://user-images.githubusercontent.com/3472541/37465572-c8784d0e-285b-11e8-9fda-c1bc9df7e465.png">



When I open the page source the element exist but he's not visible, here is the line in question : 
```
<XCUIElementTypeStaticText type="XCUIElementTypeStaticText" value="Gel douche corps et cheveux vitaminé" name="Gel douche corps et cheveux vitaminé" label="Gel douche corps et cheveux vitaminé" enabled="true" visible="false" x="290" y="909" width="110" height="36" />
```

**Comment by jojo91** [comment]
Is that possible to create a new condition operation that will be "IfElementIsVisible" ?
